src/Blazored.FluentValidation/PropertyPathHelper.cs

```diff
@@ -0,0 +1,111 @@
+using System.Collections;
+using System.Reflection;
+using Microsoft.AspNetCore.Components.Forms;
+
+namespace Blazored.FluentValidation;
+
+internal static class PropertyPathHelper
+{
+    private class Node
+    {
+        public Node? Parent { get; set; }
+        public object? ModelObject { get; set; }
+        public string? PropertyName { get; set; }
+        public int? Index { get; set; }
+    }
+
+    public static string ToFluentPropertyPath(EditContext editContext, FieldIdentifier fieldIdentifier)
+    {
+        var nodes = new Stack<Node>();
+        nodes.Push(new Node()
+        {
+            ModelObject = editContext.Model,
+        });
+
+        while (nodes.Any())
+        {
+            var currentNode = nodes.Pop();
+            var currentModelObject = currentNode.ModelObject;
+
+            if (currentModelObject == fieldIdentifier.Model)
+            {
+                return BuildPropertyPath(currentNode, fieldIdentifier);
+            }
+            
+            var nonPrimitiveProperties = currentModelObject?.GetType()
```

Comment:
I had an issue with this bit of code, in a model it kept adding DateTime types on the NodeStack in an endless loop, eventually ending in out of memory.  I think additional type guards are needed here for types such as `DateTime`.  This change to check for additional simple types fixed the issue with DateTime not being a primitive. 
 https://github.com/jafin/FluentValidation/commit/2de01ab11cce79bdc5d2fc21e76a25724d325213